packages/compass-components/src/components/accordion.tsx

```diff
@@ -50,16 +50,30 @@ const buttonHintStyles = css({
 interface AccordionProps extends React.HTMLProps<HTMLButtonElement> {
   text: string | React.ReactNode;
   hintText?: string;
+  open?: boolean;
+  setOpen?: (newValue: boolean) => void;
 }
 function Accordion({
   text,
   hintText,
+  open: _open,
+  setOpen: _setOpen,
   ...props
 }: React.PropsWithChildren<AccordionProps>): React.ReactElement {
   const darkMode = useDarkMode();
-  const [open, setOpen] = useState(false);
-  const regionId = useId('region-');
-  const labelId = useId('label-');
+  const [localOpen, setLocalOpen] = useState(_open ?? false);
+  const setOpenRef = useRef(_setOpen);
+  setOpenRef.current = _setOpen;
+  const onOpenChange = useCallback(() => {
+    setLocalOpen((prevValue) => {
+      const newValue = !prevValue;
+      setOpenRef.current?.(newValue);
+      return newValue;
+    });
+  }, []);
```

Comment:
if I understand correctly, the localOpen is just to avoid having 'open' as a callback dependency? is that worth it since the component anyway re-renders when open changes? am I missing something?

Reply:
ah I see, thanks for the explanation!